atomic_reactor/plugins/exit_koji_import.py

```diff
@@ -633,3 +494,250 @@ def run(self):
                        json.dumps(build_info, sort_keys=True, indent=4))
 
         return build_id
+
+
+class KojiImportPlugin(KojiImportBase):
+
+    key = PLUGIN_KOJI_IMPORT_PLUGIN_KEY  # type: ignore
+
+    def __init__(self, tasker, workflow,
+                 kojihub=None, url=None,
+                 verify_ssl=True, use_auth=True,
+                 koji_ssl_certs=None, koji_proxy_user=None,
+                 koji_principal=None, koji_keytab=None,
+                 blocksize=None,
+                 target=None, poll_interval=5):
+        super(KojiImportPlugin, self).__init__(tasker, workflow,
+                                               kojihub, url,
+                                               verify_ssl, use_auth,
+                                               koji_ssl_certs, koji_proxy_user,
+                                               koji_principal, koji_keytab,
+                                               blocksize,
+                                               target, poll_interval)
+
+    def set_media_types(self, extra, worker_metadatas):
+        media_types = []
+
+        # Set media_types for the base case
+        super(KojiImportPlugin, self).set_media_types(extra, worker_metadatas)
+        # Adjust media_types to include annotations
+        for platform in worker_metadatas:
+            annotations = get_worker_build_info(self.workflow,
+                                                platform).build.get_annotations()
+            if annotations.get('media-types'):
+                media_types = json.loads(annotations['media-types'])
+                break
+
+        # Copy existing; append with new, if any; de-dupe and re-sort
+        extra_image_mts = copy.deepcopy(extra['image'].get('media_types', []))
+        # FWIW: `set([].extend([]))` (the case where both
+        #     `extra['image']['media_types']` *and* `media_types` are `[]`)
+        #     throws "'NoneType' object is not iterable"
+        extra_image_mts.extend(media_types)
+        if media_types and (type(extra_image_mts) == list):
+            extra['image']['media_types'] = sorted(set(extra_image_mts))
```

Comment:
The comment is not exactly accurate IMO. This is how `extend()` works in python:
```python
>>> some_list = [1, 2, 3]
>>> some_other_list = [4, 5, 6]
>>> print(some_list.extend(some_other_list))
None
>>> print(some_list)
[1, 2, 3, 4, 5, 6]
```

`extend()` only modifies `some_list`, it does not return anything - i.e. in Python terms: returns `None`

There should also be no need to check the type of `extra_image_mts`

Reply:
(；⌣̀_⌣́) see below...